src/rust/engine/src/externs/address.rs

```diff
@@ -680,9 +680,14 @@ impl Address {
         Ok(format!("{prefix}{path}{target}{generated}{params}"))
     }
 
-    fn parametrize(&self, parameters: BTreeMap<String, String>) -> Self {
-        let mut merged_parameters = self.parameters.clone();
-        merged_parameters.extend(parameters);
+    fn parametrize(&self, parameters: BTreeMap<String, String>, replace: bool) -> Self {
```

Comment:
replace is required here, should default to `false`.

Reply:
(this is causing the failing test in CI..)

Reply:
we should call this out in the notes then, as this is a change in behavior for the `Address` type that may be used by plugins.

Reply:
(marking as unresolved until we have updated notes regarding the plugin api change)

Reply:
Oh I didn't think about it being a public API, would you rather me to backtrack and keep the `replace: bool=False`?

Reply:
I preemptively added it back, but I can remove it if you prefer to change the API and add a note.

Reply:
keeping it backwards compatible feels less risky. Thanks!